sbat.c

```diff
@@ -335,4 +335,75 @@ parse_sbat_var(list_t *entries)
 	FreePool(data);
 	return EFI_INVALID_PARAMETER;
 }
+
+EFI_STATUS
+set_sbat_uefi_variable(void)
+{
+	EFI_STATUS efi_status = EFI_SUCCESS;
+	UINT32 attributes = 0;
+
+	CHAR8 *sbat = NULL;
+	UINTN sbatsize = 0;
+
+	/* get size */
+	efi_status = gRT->GetVariable(L"SBAT", &SHIM_LOCK_GUID, &attributes,
+			&sbatsize, sbat);
+	sbat = AllocateZeroPool(sbatsize);
```

Comment:
Sorry, missed before but you need to check for allocation failures here and return `EFI_OUT_OF_RESOURCES`.